prometheus-ksonnet/grafana/datasources.libsonnet

```diff
@@ -25,6 +25,29 @@
     },
   },
 
+  datasource+:: {
+    new(name, url, type, default=false, method='GET'):: {
+      name: name,
+      type: type,
+      access: 'proxy',
+      url: url,
+      isDefault: default,
+      version: 1,
+      editable: false,
+      jsonData: {
+        httpMethod: method,
+      },
+    },
+    withBasicAuth(username, password):: {
+      basicAuth: true,
+      basicAuthUser: username,
+      basicAuthPassword: password,
+    },
+    withJsonData(data):: {
```

Comment:
```suggestion
    withJsonData(data, method='GET'):: {
```
?

Reply:
I've achieved this by removing it from the `new()` method and adding a `withHttpMethod()` function.